torchvision/prototype/models/_utils.py

```diff
@@ -1,13 +1,55 @@
+import functools
 import warnings
-from typing import Any, Dict, Optional, TypeVar
+from typing import Any, Dict, Optional, TypeVar, Callable, Tuple, Union
 
-from ._api import WeightsEnum
+from torch import nn
+from torchvision.prototype.utils._internal import kwonly_to_pos_or_kw
 
+from ._api import WeightsEnum
 
 W = TypeVar("W", bound=WeightsEnum)
+M = TypeVar("M", bound=nn.Module)
 V = TypeVar("V")
 
 
+def handle_legacy_interface(
+    **pretrained_weights: Tuple[str, Union[Optional[W], Callable[[Dict[str, Any]], Optional[W]]]]
+):
+    def outer_wrapper(builder: Callable[..., M]) -> Callable[..., M]:
+        @functools.wraps(builder)
+        def inner_wrapper(**kwargs: Any) -> M:
+            for weights_param, (pretrained_param, default) in pretrained_weights.items():  # type: ignore[union-attr]
+                weights_arg = kwargs.get(weights_param)
+                if weights_param in kwargs and not isinstance(weights_arg, bool):
+                    continue
+
+                pretrained_positional = pretrained_param not in kwargs
+                pretrained_arg = weights_arg if pretrained_positional else kwargs.pop(pretrained_param)
+                if pretrained_arg:
+                    default_arg = default(kwargs) if callable(default) else default
+                    if default_arg is None:
+                        raise ValueError(f"No checkpoint is available for model {builder.__name__}")
+                else:
+                    default_arg = None
+
+                msg = f"The current behavior is equivalent to passing `{weights_param}={default_arg}`."
+                if not pretrained_positional:
+                    msg = (
+                        f"The parameter '{pretrained_param}' is deprecated, please use '{weights_param}' instead. {msg}"
+                    )
+                if default_arg is not None:
+                    msg = f"{msg} You can also use `{weights_param}='default'` to get the most up-to-date weights."
```

Comment:
Thought strings are supported, they are not the preferred way to do it. Instead of `'default'` suggest to use `Some_Weights.default`